### PR TITLE
Web UI for the JSON encoder 

### PR DESCRIPTION
**Pull request by neysofu**
Here's a fully functional, albeit minimal, web UI for encoding EBO payloads.

- `wasm-bindgen` for automatically generating WASM bindings to `json-oracle-encoder`.
- [`monaco-editor`](https://github.com/microsoft/monaco-editor) for the code editor UI, using the `monaco-editor-webpack-plugin`.
- https://github.com/JamesIves/github-pages-deploy-action `@v4` for automatic deployment.
- [Bulma](https://bulma.io) for basic CSS styling.

The line count of the PR is quite scary, but it's mostly due to `package-lock.json` and `Cargo.lock`.

Deployment to GH Pages happens every time we push to `main`.

### CODE REVIEW

**Comment by neysofu** [comment]
Closes https://github.com/graphprotocol/block-oracle/issues/197.

**Comment by neysofu** [comment]
@tilacog you can test locally with `cd crates/json-encoder-web && npm install && npm run serve`

File: packages/json-encoder-web/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	minimap: {
+		enabled: false
+	},
+	theme: 'vs-light'
+});
+
+document.getElementById('compile-button').onclick = function () {
+	console.log('button was clicked');
+	let input = editor.getValue();
+	let compiled = wasm.compile(input, true);
```
Comment (tilacog) [inline packages/json-encoder-web/index.ts:35]: Currently, the form produces the output for `Calldata`, regardless of which option is selected.

I believe the fix revolves around this `true` value, considering whether the `Calldata` or `Bytes payload` option is selected instead.

**Review by tilacog — CHANGES_REQUESTED** [review]
This is awesome!

File: packages/json-encoder-web/index.ts (hunk 1 above)
Comment (neysofu) [reply packages/json-encoder-web/index.ts]: Right, I forgot about that missing piece. Fixed in https://github.com/graphprotocol/block-oracle/pull/205/commits/d5f225e1fcb6287ae9129e596a5ba1cb6e324744.

**Pull request merged**